c10/core/CPUCachingAllocator.cpp

```diff
@@ -0,0 +1,105 @@
+#include <c10/core/CPUCachingAllocator.h>
+
+namespace c10 {
+
+namespace {
+thread_local CPUCachingAllocator* caching_allocator_ptr{nullptr};
```

Comment:
Hey, sorry I'm late to the game. Do you know if `thread_local` here compiles to a `__tls_get_addr` on mobile? On recent profiling we've been doing for core overheads, we've noticed that `__tls_get_addr` is quite expensive and so it's a bit surprising to see this here. (thread local doesn't always compile to this, it's only in certain situations, so I'm wondering if this applies to mobile too.)